lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +151,10 @@ def install
       end
     end
 
+    if @resource[:package_arch]
+      wanted = "#{wanted}.#{@resource[:package_arch]}"
+    end
+
```

Comment:
This code path is during install, should we be able to uninstall an arch specific package too?